api/pkg/imagebuilder/imagebuilder.go

```diff
@@ -615,46 +620,75 @@ func (c *imageBuilder) createKanikoJobSpec(
 		fmt.Sprintf("--dockerfile=%s", baseImageTag.DockerfilePath),
 		fmt.Sprintf("--context=%s", baseImageTag.BuildContextURI),
 		fmt.Sprintf("--build-arg=%s=%s", baseImageEnvKey, baseImageTag.ImageName),
+		fmt.Sprintf("--build-arg=%s=%s", modelArtifactStorageTypeKey, c.artifactService.GetType()),
```

Comment:
This argument gets passed to the batch predictor/pyfunc server image, and is used to determine the correct CLI (GCS or AWS) to use to pull the model artifacts from.